Feat: weather api call

dmdwn99 opened the pull request:
## 📕 제목
- #11 

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] 작업내용1: HourlyWeather update 메서드 추가
- [x] 작업내용2: HourlyWeatherDto 생성
- [x] 작업내용3: OpenWeatherMap One call Api 호출 및 HourlyWeather 저장

## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

dmdwn99 committed:
Feat: OpenWeatherMap API 요청 구현 #11

1. HourlyWeather update 메서드 추가
2. HourlyWeatherDto 생성
3. OpenWeatherMap One call Api 호출 및 HourlyWeather 저장

dmdwn99 committed:
Merge branch 'develop' into feat/weather-api-call

# Conflicts:
#	build.gradle

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+    private final RegionRepository repository;
+
+    private final String key = "ebef73b5168e228a84f39f3bbe8ea6da";
+    private final String url = "https://api.openweathermap.org/data";
```
key랑 url은 @Value로 빼야할 것 같아요. 노출돼서 재발급 받아야 할 것 같네요? ㅎ 재발급 받고 secret 수정해야 돼서 회의 때 같이 맞춰 보아요.

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+        stringBuilder.append(url+"/3.0/onecall?");
+        stringBuilder.append("lat="+region.getLatitude()+"&lon="+region.getLongitude());
+        stringBuilder.append("&units=metric&exclude=current,minutely,alerts&appid="+key);
+        String url = stringBuilder.toString();
```
노란 줄 뜨는 건 바꿔주는 게 좋을 것 같아요. 노란 줄 내용은 'StringBuilder stringBuilder' can be replaced with 'String' 이었는데, 이 부분은 굳이 StringBuilder를 쓰지 않아도 되고, String으로 대체하는 게 좋다는 소리 같네용

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
+//        saveDailyWeather(jsonString);
```
// TODO 하고 주석처리하면 좋을 것 같습니다~

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+        try {
+            saveHourlyWeather(jsonString, region);
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
```
RuntimeException 말고 저희 parse 실패 에러코드 만들어서 커스텀 익셉션 사용하면 좋을 것 같아요. `throw new CustomException(ErrorCode.ㅇㅇㅇ)`

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+        JSONParser jsonParser = new JSONParser();
+        JSONObject object = (JSONObject) jsonParser.parse(jsonString);
+        JSONArray hourlyWeatherInfo = (JSONArray) object.get("hourly");
+        for (int i=0;i<hourlyWeatherInfo.size();i++) {
```
여기도 노란 줄( 'for' loop can be replaced with enhanced 'for' ) 뜨네용

reeruryu commented on src/main/java/com/seungah/todayclothes/service/WeatherService.java:
```diff
@@ -0,0 +1,88 @@
+                    .temp((Double) object.get("temp"))
+                    .humidity((Double) object.get("humidity"))
+                    .icon((String) iconObject.get("icon"))
+                    .build();
```
이해가 안돼서 그런데, 왜 dto를 만든 건가요??(그냥 바로 save하면 되지 않나 싶어서요) 그리고 저장을 안하는 거 같은데 아직 구현 안한 건가요??!

reeruryu replied:
추가로 ClassCastException 뜨는데 확인 부탁드립니다~!

reeruryu commented on src/main/java/com/seungah/todayclothes/entity/Region.java:
```diff
@@ -0,0 +1,26 @@
+	private String name;
+
+	private String latitude;
+	private String longitude;
```
그때 왜 Double로 안 받고 String으로 받는지 말한 거 같은데 기억이 안나서요.. 댓글로 다시 한 번 설명 부탁 드려요 ㅠㅠ

reeruryu commented on src/main/java/com/seungah/todayclothes/entity/HourlyWeather.java:
```diff
@@ -25,13 +22,19 @@ public class HourlyWeather extends BaseEntity {
+	@JoinColumn(name = "region_id")
+	private Region region;
+
+	public void update(Double temp, Double humidity, String icon){
```
이 부분 사용 안되고 있는 거 같은데 왜 만든 걸까요??

reeruryu reviewed (changes requested):
응주 고생했어 👍👍👍 
내일 회의 때 얘기해 봅시다 !

dmdwn99 committed:
Feat: OpenWeatherMap API 요청 구현 (오류 수정) #16

1. Weather DB 수정을 위한 from 메서드 생성
2. 날짜 type LocalDateTime으로 변경
3. 날짜와 지역을 통한 DB 조회
4. Region Enum Type에서 Class로 변경

dmdwn99 committed:
Feat: OpenWeatherMap API 요청 구현 (오류 수정) #16

1. WeatherService 누락으로 다시 Commit
